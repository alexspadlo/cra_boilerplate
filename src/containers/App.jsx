import React from 'react'
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom'
import PublicRoute from '../components/RouterComponents/PublicRoute/PublicRoute'
import PrivateRoute from '../components/RouterComponents/PrivateRoute/PrivateRoute'
import HomePage from '../components/HomePage/HomePage'
import Dashboard from '../components/Dashboard/Dashboard'

class App extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <PublicRoute
            exact
            path='/'
            component={HomePage}
          />
          <PrivateRoute
            exact
            path='/dashboard'
            component={Dashboard}
          />
          <Redirect from="*" to="/" />
        </Switch>
      </Router>
    )
  }
}

export default App