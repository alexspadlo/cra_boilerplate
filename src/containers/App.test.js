import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

describe('Testing App Container', () => {
  test('Render without props and match snapshot', () => {
    const container = render(<App />)
    expect(container).toMatchSnapshot()
  })
})