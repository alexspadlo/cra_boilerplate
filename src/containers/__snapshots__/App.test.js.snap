// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Container Render without props and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container p-0"
      >
        <div
          class="row p-5"
        >
          <div
            class="col text-center m-auto"
          >
            <h1
              class="text-success pb-4"
            >
              <strong>
                HEY !
              </strong>
            </h1>
            <button
              class="text-center outButton"
            >
              Log in
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container p-0"
    >
      <div
        class="row p-5"
      >
        <div
          class="col text-center m-auto"
        >
          <h1
            class="text-success pb-4"
          >
            <strong>
              HEY !
            </strong>
          </h1>
          <button
            class="text-center outButton"
          >
            Log in
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
